front/components/agent_builder/capabilities/shared/ProcessingMethodSection.tsx

```diff
@@ -118,14 +119,30 @@ export function ProcessingMethodSection() {
   }, [mcpServerViewsWithKnowledge, sources.in, mcpServerView]);
 
   useEffect(() => {
-    // Update mcpServerView only if it's null and we have servers to display
-    if (serversToDisplay && mcpServerView === null) {
-      const [defaultServer] = serversToDisplay;
-      if (defaultServer) {
-        setValue("mcpServerView", defaultServer, { shouldDirty: false });
+    if (serversToDisplay && sources.in.length > 0 && !mcpServerView) {
+      const allTablesOrDatabases = sources.in.every(
+        isRemoteDatabaseOrTableItem
+      );
+
+      if (allTablesOrDatabases) {
+        const tableQueryServer = serversToDisplay.find(
```

Comment:
We should select v2 if they are available? I had several tickets from customers that table query v1 didn't work and enabled v2 for them